src/compressed_tensors/compressors/marlin_24.py

```diff
@@ -0,0 +1,267 @@
+from compressed_tensors.quantization import QuantizationArgs, QuantizationStrategy
+from compressed_tensors.quantization.lifecycle.forward import quantize
+from compressed_tensors.utils import is_quantization_param, merge_names
+from torch import Tensor
+from tqdm import tqdm
+
+
+_LOGGER: logging.Logger = logging.getLogger(__name__)
+
+
+@Compressor.register(name=CompressionFormat.marlin_24.value)
+class Marlin24Compressor(Compressor):
+    """
+    Compresses a quantized model with 2:4 sparsity structure for inference with the
+    Marlin24 kernel. Decompression is not implemented for this compressor.
+    """
+
+    COMPRESSION_PARAM_NAMES = ["weight_packed", "scale_packed", "meta"]
+
+    @staticmethod
+    def validate_quant_compatability(model_quant_args: Dict[str, QuantizationArgs]):
+        """
+        Checks if every quantized module in the model is compatible with Marlin24
+        compression. Quantization must be channel or group strategy with group_size
+        of 128. Only symmetric quantization is supported
+
+        :param model_quant_args: dictionary of mapping module names to their
+            quantization configuration
+        :return: True if all modules are compatible with Marlin24 compression, raises
+            a ValueError otherwise
+        """
+        for name, quant_args in model_quant_args.items():
+            strategy = quant_args.strategy
+            group_size = quant_args.group_size
+            symmetric = quant_args.symmetric
+            if (
+                strategy is not QuantizationStrategy.GROUP
+                and strategy is not QuantizationStrategy.CHANNEL
+            ):
+                raise ValueError(
+                    f"Marlin24 Compressor is only valid for group and channel "
+                    f"quantization strategies, got {strategy} in {name}"
+                )
+
+            if group_size is not None and group_size != 128:
+                raise ValueError(
+                    f"Marlin24 Compressor is only valid for group size 128, "
+                    f"got {group_size} in {name}"
+                )
+
+            if not symmetric:
+                raise ValueError(
+                    f"Marlin24 Compressor is only valid for symmetric quantzation, "
+                    f"got symmetric={symmetric} in {name}"
+                )
+
+        return True
+
+    @staticmethod
+    def validate_sparsity_structure(
+        name: str, weight: Tensor, num_rows_to_sample: int = 20
+    ) -> bool:
+        """
+        Checks if a tensor fits the 2:4 sparsity structure by sampling a specified
+        number of rows.
+
+        :param name: name of the tensor to check
+        :param weight: tensor to check for sparsity structure
+        :param num_rows_to_sample: number of rows to check the sparsity structure of
+        :return: True if all sampled rows match the 2:4 sparsity structure, raises
+            ValueError otherwise
+        """
+        BLOCK_SIZE = 4
```

Comment:
we have similar help function (by @rahul-tuli ) in SparseML: https://github.com/neuralmagic/sparseml/blob/main/src/sparseml/transformers/compression/helpers.py#L30

How about we move this functionality to compressed-tensors and then import to SparseML if needed?

Reply:
I think this code will be useful for other features. WIll be easier to test if its an isolated func. The sparseml ref code above is quite similar with 2:4 here. might be ok to build on top
 
Instead of having N many validations, i would just make a single validate func that  would check both the validate_quant_compatability and validate_sparsity_structure, importing relevant helpers from a common importing directory.

Reply:
Nice didn't realize that existed. The benefit of this implementation though is we only sample a few rows so it should be quicker than the SparseML implementation. Let me swap them out and see how much of a speed hit it is